nikola/plugins/compile/rest/__init__.py

```diff
@@ -59,6 +59,38 @@ class CompileRest(PageCompiler):
     demote_headers = True
     logger = None
 
+    def read_metadata(self, post, file_metadata_regexp=None, unslugify_titles=False, lang=None):
+        """Read the metadata from a post, and return a metadata dict."""
+        if lang is None:
+            lang = LocaleBorg().current_lang
+        source = post.translated_source_path(lang)
+
+        with io.open(source, 'r', encoding='utf-8') as inf:
+            document = docutils.core.publish_doctree(
```

Comment:
Yeah to both things. We can probably keep the doctree somewhere in the Post object and reuse it if it's there.

Same about markdown, actually.